src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -129,6 +125,34 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M> {
                     self.binop_with_overflow(bin_op, lhs, rhs, dest)?;
                 }
             }
+            "unchecked_shl" | "unchecked_shr" => {
+                let bits = dest.layout.size.bytes() as u128 * 8;
+                let l = self.read_value(args[0])?;
+                let r = self.read_value(args[1])?;
+                let r_ty = substs.type_at(0);
+                let r_layout_of = self.layout_of(r_ty)?;
+                let r_val =  r.to_scalar()?.to_bits(r_layout_of.size)?;
+                let bin_op = match intrinsic_name {
+                    "unchecked_shl" => {
+                        if r_val >= bits {
```

Comment:
this check can be moved out of the match on the intrinsic name

Reply:
Not necessarily: The error message is a different one (`shl` vs `shr`).

Reply:
We're calling `format` anyway, so you can add the `intrinsic_name` as another formatting parameter